paper-linear-progress.html

```diff
@@ -0,0 +1,916 @@
+          value: 100
+        },
+
+        /**
+         * When set, progress bars animate across the element, expressing an operation which takes
+         * an unknown amount of time.
+         */
+        indeterminate: {
+          type: Boolean,
+          observer: '_indeterminateChanged',
+          // We give a default value so that correct ARIA attributes are applied when element is
+          // initialized.
+          value: false
+        },
+
+        /**
+         * When set, will reverse all animations, transforms, and transitions occuring within the
+         * element. Useful when working with RTL languages, as well as distinguishing between phases
+         * of an operation.
+         */
+        reversed: {
+          type: Boolean,
+          observer: '_reversedChanged'
+        },
+
+        /**
+         * When set, will display buffering dots which appear to animate across the element. Useful
+         * when displaying progress for operations involving media, during which content may need to
+         * be buffered.
+         */
+        buffering: {
+          type: Boolean,
+          observer: '_bufferingChanged'
+        },
+
+        /**
+         * When set, the linear progress element will be hidden from view. Note that this is
+         * _different_ from `hidden`, as it does not set `display: none;` but rather just changes
+         * the element's opacity.
+         */
+        closed: {
+          type: Boolean,
+          observer: '_closedChanged'
+        },
+
+        /**
+         * Animation configuration for NeonAnimationRunnerBehavior.
+         */
+        animationConfig: {
+          value: function() {
+            // NOTE(traviskaufman): This is initialized within the attached() method, once we know
+            // the height of the progress indicator.
+            return {};
+          }
+        },
+
+        __needsBufferingAnimationOnAttached: {
+          // HACK: We want to have buffering play an animation when it's set, but we can't
+          // animate unless we first have the computed height of the element, which we can only
+          // retrieve once the element's attached. Therefore, we set up a queue of animations that
+          // need to be run, and if we encounter any animations. Note that this can be fixed once
+          // https://github.com/Polymer/polymer/issues/2399 is resolved. Then, we can do all
+          // animations within CSS.
+          type: Boolean,
+          value: false
+        }
+      },
+
+      hostAttributes: {
+        role: 'progressbar'
+      },
+
+      ready: function() {
+        this.toggleClass('isIE', /MSIE/.test(navigator.userAgent));
+      },
+
+      attached: function() {
+        this._applyBufferingDotsBackground();
+        this._applyAnimationConfig();
+        if (this.__needsBufferingAnimationOnAttached) {
+          this._playCorrectBufferingAnimation();
+        }
+      },
+
+      _valueChanged: function(newValue) {
```

Comment:
Our convention for methods is single underscore prefix (`_`) for protected members and double underscore prefix (`__`) for private members. We try to provide documentation for all public and protected implementation. If you have reason to believe that there would be no reason to override any of these protected methods, feel free to change their signature to make them private in lieu of documenting them.